### PR TITLE
feat: Add fsx filesystems

### LINKED ISSUE #1198: FSx: file-systems
### Describe the resource.

https://docs.aws.amazon.com/cli/latest/reference/fsx/describe-file-systems.html

### Use Case

Describe FSx file-systems.

### Additional context

_No response_

### LINKED ISSUE #1375: FSx: file-systems
### Describe the resource.

https://docs.aws.amazon.com/cli/latest/reference/fsx/describe-file-systems.html

### Use Case

Describe FSx file-systems.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by irmatov**
Closes cloudquery/cloudquery#1375

### CODE REVIEW

File: docs/tables/aws_fsx_filesystem_lustre_configuration.md
Comment (disq) [inline docs/tables/aws_fsx_filesystem_lustre_configuration.md:12]: `data_repository_configuration_` => `data_repo_cfg_`?

Comment (irmatov) [reply docs/tables/aws_fsx_filesystem_lustre_configuration.md]: Renamed.

File: resources/services/fsx/filesystems.go
Comment (roneli) [inline resources/services/fsx/filesystems.go:58]: maybe id?

File: resources/services/fsx/filesystems.go
Comment (roneli) [inline resources/services/fsx/filesystems.go:68]: maybe remove file_system prefix in file_system table?

File: resources/services/fsx/filesystems.go
Comment (roneli) [inline resources/services/fsx/filesystems.go:237]: timestamp?

File: resources/services/fsx/filesystems.go
Comment (roneli) [inline resources/services/fsx/filesystems.go:282]: cidr?

File: resources/services/fsx/filesystems.go
Comment (roneli) [inline resources/services/fsx/filesystems.go:293]: IP Array?

File: resources/services/fsx/filesystems.go
Comment (roneli) [inline resources/services/fsx/filesystems.go:306]: IP Array?

File: resources/services/fsx/filesystems.go
Comment (roneli) [inline resources/services/fsx/filesystems.go:472]: ip array?

**Review by roneli — CHANGES_REQUESTED** [review]
LGTM, had a few nits on types, if they are complicated to check verify/not sure then skip them, better to keep original type than have errors rolling out.

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:58]: Renamed

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:68]: Done

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:237]: I don't think so, as it is just a time of day, not a specific point in time.

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:326]: Same as above, I don't think it's a timestamp. More like a time of day but I don't see a suitable type in SDK.

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:359]: same ^

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:391]: same ^

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:507]: same as above

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:510]: converted to json.

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:285]: can't verify what the format is, so I would keep it as a string for now

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:296]: yes, but can't convert it now due to cq-gen quirk/ bug

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:309]: yes, but can't convert it now due to cq-gen quirk/ bug

File: resources/services/fsx/filesystems.go
Comment (irmatov) [reply resources/services/fsx/filesystems.go:480]: can't due to cq-gen

File: resources/services/fsx/filesystems.go
Comment (bbernays) [reply resources/services/fsx/filesystems.go:296]: I don't think we should be prioritizing getting the data in the ideal format, rather than optimizing for what cq-gen will allow for. Users shouldn't be impacted by our automation tooling

**Review by roneli — APPROVED** [review]
LGTM, let me check the cq-gen bug quick before we merge this later today

**Pull request closed without merging**
